assimilation_code/modules/assimilation/assim_tools_mod.f90

```diff
@@ -1242,40 +772,10 @@ subroutine filter_assim(ens_handle, obs_ens_handle, obs_seq, keys,           &
 call get_close_destroy(gc_state)
 call get_close_destroy(gc_obs)
 
-! print some stats about the assimilation
-! (if interesting, could print exactly which obs # was fastest and slowest)
-if (my_task_id() == 0 .and. timing(MLOOP)) then
-   write(msgstring, *) 'average assim time: ', sum(elapse_array) / size(elapse_array)
-   call error_handler(E_MSG,'filter_assim:',msgstring)
-
-   write(msgstring, *) 'minimum assim time: ', minval(elapse_array)
-   call error_handler(E_MSG,'filter_assim:',msgstring)
-
-   write(msgstring, *) 'maximum assim time: ', maxval(elapse_array)
-   call error_handler(E_MSG,'filter_assim:',msgstring)
-endif
-
-if (timing(MLOOP)) deallocate(elapse_array)
-
 ! do some stats - being aware that unless we do a reduce() operation
 ! this is going to be per-task.  so only print if something interesting
 ! shows up in the stats?  maybe it would be worth a reduce() call here?
 
-!>@todo FIXME:  
-!  we have n_close_obs_items and n_close_state_items for each assimilated
-!  observation.  what we really want to know is across the tasks is there
-!  a big difference in counts?  so that means communication.  maybe just
-!  the largest value?  and the number of 0 values?  and if the largest val
-!  is way off compared to the other tasks, warn the user?
-!  we don't have space or time to do all the obs * tasks but could we
-!  send enough info to make a histogram?  compute N bin counts and then
-!  reduce that across all the tasks and have task 0 print out?
-! still thinking on this idea.
-!   write(msgstring, *) 'max state items per observation: ', maxval(n_close_state_items)
-!   call error_handler(E_MSG, 'filter_assim:', msgstring)
-! if i come up with something i like, can we use the same idea
-! for the threed_sphere locations boxes?
-
 ! Assure user we have done something
 if (print_trace_details >= 0) then
 write(msgstring, '(A,I8,A)') &
```

Comment:
Lots of the caching mess is still left in this module. Agree that it would be cleaner in the long-run to push it off towards the location modules, but not doing it now.